ecs-cli/modules/config/readwriter_v1.go

```
@@ -66,48 +67,140 @@ func readINI(dest *Destination) (*CLIConfig, map[interface{}]interface{}, error)
+		}
+	}
+	profile, ok := profileMap["ecs_profiles"].(map[interface{}]interface{})[profileKey].(map[interface{}]interface{})
+	if !ok {
+		return errors.New("Format issue with profile config file; expected key not found.")
+	}
+
+	configMap[awsAccessKey] = profile[awsAccessKey]
+	configMap[awsSecretKey] = profile[awsSecretKey]
+	if cliConfig.AWSAccessKey, ok = profile[awsAccessKey].(string); !ok {
+		return errors.New("Format issue with profile config file; expected key not found.")
+	}
+	if cliConfig.AWSSecretKey, ok = profile[awsSecretKey].(string); !ok {
+		return errors.New("Format issue with profile config file; expected key not found.")
+	}
+
+	return nil
+
+}
+
+func processClusterMap(clusterConfigKey string, clusterMap map[interface{}]interface{}, configMap map[interface{}]interface{}, cliConfig *CLIConfig) error {
+	if clusterConfigKey == "" {
+		var ok bool
+		if clusterConfigKey, ok = clusterMap["default"].(string); !ok {
+			return errors.New("Format issue with cluster config file; expected key not found.")
+		}
+	}
+	cluster, ok := clusterMap["clusters"].(map[interface{}]interface{})[clusterConfigKey].(map[interface{}]interface{})
+	if !ok {
+		return errors.New("Format issue with cluster config file; expected key not found.")
+	}
+
+	configMap[clusterKey] = cluster[clusterKey]
+	configMap[regionKey] = cluster[regionKey]
+	if cliConfig.Cluster, ok = cluster[clusterKey].(string); !ok {
+		return errors.New("Format issue with cluster config file; expected key not found.")
+	}
+	if cliConfig.Region, ok = cluster[regionKey].(string); !ok {
+		return errors.New("Format issue with cluster config file; expected key not found.")
+	}
+
+	// Prefixes
+	// ComposeProjectNamePrefix
+	if _, ok := cluster[composeProjectNamePrefixKey]; ok {
+		configMap[composeProjectNamePrefixKey] = cluster[composeProjectNamePrefixKey]
+		if cliConfig.ComposeProjectNamePrefix, ok = cluster[composeProjectNamePrefixKey].(string); !ok {
```

You should throw this error first. You are checking for the same condition as the previous if statement.